vic/drivers/shared_image/src/vic_init_output.c

```diff
@@ -339,36 +386,107 @@ initialize_history_file(nc_file_struct *nc)
     // define the netcdf variable latitude
     status = nc_def_var(nc->nc_id, global_domain.info.lat_var, NC_DOUBLE, ndims,
                         dimids, &(lat_var_id));
-    if (status != NC_NOERR) {
-        log_err("Error defining lat variable in %s", nc->fname);
-    }
+    check_nc_status(status, "Error defining lat variable in %s",
+                    stream->filename);
     status = nc_put_att_text(nc->nc_id, lat_var_id, "long_name",
                              strlen("latitude"), "latitude");
-    if (status != NC_NOERR) {
-        log_err("Error adding attribute in %s", nc->fname);
-    }
+    check_nc_status(status, "Error adding latitude long_name attribute in %s",
+                    stream->filename);
     status = nc_put_att_text(nc->nc_id, lat_var_id, "units",
                              strlen("degrees_north"), "degrees_north");
-    if (status != NC_NOERR) {
-        log_err("Error adding attribute in %s", nc->fname);
-    }
+    check_nc_status(status, "Error adding latitude units attribute in %s",
+                    stream->filename);
     status = nc_put_att_text(nc->nc_id, lat_var_id, "standard_name",
                              strlen("latitude"), "latitude");
-    if (status != NC_NOERR) {
-        log_err("Error adding attribute in %s", nc->fname);
+    check_nc_status(status,
+                    "Error adding latitude standard_name attribute in %s",
+                    stream->filename);
+
+    // create output variables
+    for (j = 0; j < stream->nvars; j++) {
+        varid = stream->varid[j];
+
+        set_nc_var_dimids(varid, nc, &(nc->nc_vars[j]));
+
+        // define the variable
+        status = nc_def_var(nc->nc_id,
+                            out_metadata[varid].varname,
+                            nc->nc_vars[j].nc_type,
+                            nc->nc_vars[j].nc_dims,
+                            nc->nc_vars[j].nc_dimids,
+                            &(nc->nc_vars[j].nc_varid));
+        check_nc_status(status, "Error defining variable %s in %s.  Status: %d",
+                        out_metadata[varid].varname, stream->filename, status);
+
+        // Add compression (only works for netCDF4 filetype)
+        if (stream->compress) {
+            status = nc_def_var_deflate(nc->nc_id, nc->nc_vars[j].nc_varid,
+                                        true, true, stream->compress);
+            check_nc_status(
+                status,
+                "Error setting compression level in %s for variable: %s",
+                stream->filename, out_metadata[varid].varname);
+        }
+
+        // set the fill value attribute
+        if (nc->nc_vars[j].nc_type == NC_DOUBLE) {
```

Comment:
Should probably replace with `switch ... case` statement. Just open issue and leave as is.